### ISSUE TITLE
Time limited tests: incorrect use of threads

### ISSUE THREAD

**Issue by charpov** [scalatest/scalatest#981]
The following example seems to show that:

- 1 timer (including 1 thread) is created *for each test*;
- the timers are never cancelled and prevent the JVM from terminating.

A thread dump shows 100 threads named `Timer-0`, `Timer-1`, etc.

This is a *major* issue, because a suite with a large number of tests fail from running out of threads.  This behavior is observed with Scalatest 3.0.0 but *not* with 2.2.6.

``` scala
import org.scalatest.FunSuite
import org.scalatest.concurrent.TimeLimitedTests
import org.scalatest.time.SpanSugar._

class Tests extends FunSuite with TimeLimitedTests {
  def timeLimit = 1.second

  for (i <- 1 to 100)
    test("test "+i) {
    }
}

object Main extends App {
  val suite = new Tests
  suite.execute()
}
```

**Comment by bvenners** [comment]
We did have to refactor the implementation of TimeLimitedTests when we added Scala.js support, so it is possible we lost the cancelation. We'll investigate.

**Comment by charpov** [comment]
Do you really need multiple timers?  Even if they are properly cancelled, this seems wasteful.  Can't the same timer be used to monitor multiple tests?  And if this makes the system less responsive (i.e., it takes more time to detect timeouts), can a `ScheduledExecutionService` be used, which would give some control over the number of timer threads (as opposed to one per test, which is unsustainable)?

I've downgraded to 2.2.6 for now so I can run my suites.

**Comment by bvenners** [comment]
Well TimeLimitedTests is used on both the JVM and Scala.js, so I believe we made an abstraction that is used on both. But even in prior releases of ScalaTest which was JVM only, I'm pretty sure we were creating one Timer per test. Hadn't heard of a problem with that approach.

**Comment by charpov** [comment]
You're right (I just checked).  Even with 2.2.6, a timer is created for each test.  It's a bit wasteful, but since each timer is cancelled before the next test runs, the total number of threads being used doesn't grow, so large suites can run with limited resources.

**Comment by cheeseng** [comment]
I submitted a PR for this: 

https://github.com/scalatest/scalatest/pull/982
